### PR TITLE
adding fix for local storage saving as table and as list

### PR DESCRIPTION
**Pull request by hclarkEnq**
## Description

Fixing bug in saving view type search for local storage in table and list view type preferences

## Tickets

* [CNFT1-2556](https://cdc-nbs.atlassian.net/browse/CNFT1-2556)

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

[CNFT1-2556]: https://cdc-nbs.atlassian.net/browse/CNFT1-2556?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by stevegsa — APPROVED** [review]

**Review by rkrusselenq — DISMISSED** [review]

**Review by adamloup-enquizit — COMMENTED** [review]
suggestion:  This is leaking the concern of storing the default settings into another hook.  Try adding a function to the `useSearchSettings` that saves the new local defaults.

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]
This is blurring the responsibilities of `useSearchResultDisplay` and `useSettings'.

**Comment by hclarkEnq** [comment]
> suggestion: This is leaking the concern of storing the default settings into another hook. Try adding a function to the `useSearchSettings` that saves the new local defaults.
Okay will look into this

File: apps/modernization-ui/src/apps/search/useSearchSettings.ts [hunk 1]
```diff
@@ -12,22 +12,35 @@ const defaults: SearchSettings = {
...
-            setSettings((current) => ({ ...current, defaultView: defaultView || 'table', allowToggle: true }));
+            setSettings((current) => ({ ...current, allowToggle: true }));
         }
     }, [search.view.table.enabled]);
-    return settings;
+
+    const updateDefaultView = useCallback((newView: View) => {
+        localStorage.setItem('defaultSearchView', newView);
+        setSettings((current) => ({ ...current, defaultView: newView }));
+    }, []);
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/useSearchSettings.ts:37]: question:  What is the need for `useCallback` here?

**Review by alaapbharadwaj — APPROVED** [review]

**Review by adamloup-enquizit — APPROVED** [review]
🎈 Looks good

**Pull request merged**